llvm/lib/Target/AArch64/AArch64SVEInstrInfo.td

```diff
@@ -3674,10 +3690,10 @@ let Predicates = [HasSVE2orSME] in {
   defm UABDLT_ZZZ : sve2_wide_int_arith_long<0b01111, "uabdlt", int_aarch64_sve_uabdlt>;
 
   // SVE2 integer add/subtract wide
-  defm SADDWB_ZZZ : sve2_wide_int_arith_wide<0b000, "saddwb", int_aarch64_sve_saddwb>;
-  defm SADDWT_ZZZ : sve2_wide_int_arith_wide<0b001, "saddwt", int_aarch64_sve_saddwt>;
-  defm UADDWB_ZZZ : sve2_wide_int_arith_wide<0b010, "uaddwb", int_aarch64_sve_uaddwb>;
-  defm UADDWT_ZZZ : sve2_wide_int_arith_wide<0b011, "uaddwt", int_aarch64_sve_uaddwt>;
+  defm SADDWB_ZZZ : sve2_wide_int_arith_wide<0b000, "saddwb", AArch64saddwb>;
+  defm SADDWT_ZZZ : sve2_wide_int_arith_wide<0b001, "saddwt", AArch64saddwt>;
+  defm UADDWB_ZZZ : sve2_wide_int_arith_wide<0b010, "uaddwb", AArch64uaddwb>;
+  defm UADDWT_ZZZ : sve2_wide_int_arith_wide<0b011, "uaddwt", AArch64uaddwt>;
```

Comment:
Another option (other than `performIntrinsicCombine()`) for matching the intrinsics could be to add another parameter to `sve2_wide_int_arith_wide` for the intrinsic pattern fragment so that both patterns can be registered here. I've seen this done in `sve_fp_2op_p_zd` (for example). I'm not requesting a change here; I just thought I'd mention this as I've seen it done a few times.

Reply:
OK thank you, I think I just took my current approach because a lot of the intrinsics being matched in `performIntrinsicCombine` seemed quite similar to the wide add instructions I'm adding, so it seemed worthwhile doing it in a similar way to these other intrinsics.

Reply:
Makes sense :+1: I'm not sure if there's any meaningful difference between the two approaches.

Reply:
FYI: The style in this PR is the preferred approach.  At the point where there's value in having a dedicated ISD node it's preferable to use this for all code paths (i.e. to lower the intrinsics to the same node) so that all paths can take advantage of any future DAG combines.